mycore-iiif/src/main/java/org/mycore/iiif/image/MCRIIIFImageUtil.java

```diff
@@ -39,14 +39,7 @@ public static void completeProfile(MCRIIIFImageImpl impl, MCRIIIFImageProfile pr
      * @see <a href="https://iiif.io/api/image/2.1/#uri-encoding-and-decoding">URI encoding of image identifier</a>
      */
     public static String encodeImageIdentifier(String imageIdentifier) {
-
-        try {
-            final URI uri = new URI(null, null, imageIdentifier, null, null);
-            return uri.toASCIIString().replace("/", "%2F");
-        } catch (URISyntaxException e) {
-            //99.99% this won't happen
-            throw new MCRException("Could not encode image identifier: " + imageIdentifier, e);
-        }
+        return URLEncoder.encode(imageIdentifier, StandardCharsets.UTF_8);
```

Comment:
This will not work because / needs to be encoded. Its also better to use URI encoding.

Reply:
It does work! - "/" is encoded by URLEncoder.encode().
https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html

URI.toASCIIString() did not work for me, because it interprets the image identifier as URI.
The part before a colon (":") in the identifier gets interpreted as scheme part (like http:, ...) and does not necessarily match the rules (e.g.: less than 8 characters, ...).

Since we are encoding the identifier to be used in a URL, I think URLEncoder.encode() is a good choice.

Reply:
Spaces are encoded wrong